Error when emitting typescript with typescript 6.0 installed

TS6 removed `--outFile`, so the tsd emit flag errors out.

----

**Version of emscripten/emsdk:**

5.0.4

**Failing command line in full:**
If this is compile or link-time failure please include the full failing command
along with its entire output.

```
ERROR: failed to build: failed to solve: process "/bin/sh -c emcc /tmp/hello.c -o /tmp/hello.js       -s EXPORTED_FUNCTIONS='[\"_add\"]'       -s EXPORTED_RUNTIME_METHODS='[\"cwrap\"]'       --emit-tsd /tmp/hello.d.ts" did not complete successfully: exit code: 1
```

**Full link command and output with `-v` appended:**
Even for runtime issues it helps a lot if you can include the full link command.
Adding `-v` to the link command will show all of the sub-commands run which
can help us diagnose your issue.

```
0.337  /emsdk/upstream/bin/clang -target wasm32-unknown-emscripten -fignore-exceptions -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr --sysroot=/emsdk/upstream/emscripten/cache/sysroot -DEMSCRIPTEN -Xclang -iwithsysroot/include/fakesdl -Xclang -iwithsysroot/include/compat -v -c /tmp/hello.c -o /tmp/emscripten_temp_chu_932q/hello.o
0.341 clang version 23.0.0git (https:/github.com/llvm/llvm-project e5927fecf8a6ce89e1a4eac5b828e7d42676452a)
0.341 Target: wasm32-unknown-emscripten
0.341 Thread model: posix
0.341 InstalledDir: /emsdk/upstream/bin
0.341  (in-process)
0.341  "/emsdk/upstream/bin/clang-23" -cc1 -triple wasm32-unknown-emscripten -emit-obj -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name hello.c -mrelocation-model static -mframe-pointer=none -ffp-contract=on -fno-rounding-math -mconstructor-aliases -target-cpu generic -fvisibility=hidden -debugger-tuning=gdb -fdebug-compilation-dir=/src -v -fcoverage-compilation-dir=/src -resource-dir /emsdk/upstream/lib/clang/23 -D EMSCRIPTEN -isysroot /emsdk/upstream/emscripten/cache/sysroot -internal-isystem /emsdk/upstream/lib/clang/23/include -internal-isystem /emsdk/upstream/emscripten/cache/sysroot/include/wasm32-emscripten -internal-isystem /emsdk/upstream/emscripten/cache/sysroot/include -ferror-limit 19 -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -fignore-exceptions -iwithsysroot/include/fakesdl -iwithsysroot/include/compat -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr -o /tmp/emscripten_temp_chu_932q/hello.o -x c /tmp/hello.c
0.341 clang -cc1 version 23.0.0git based upon LLVM 23.0.0git default target x86_64-unknown-linux-gnu
0.342 ignoring nonexistent directory "/emsdk/upstream/emscripten/cache/sysroot/include/wasm32-emscripten"
0.342 #include "..." search starts here:
0.342 #include <...> search starts here:
0.342  /emsdk/upstream/emscripten/cache/sysroot/include/fakesdl
0.342  /emsdk/upstream/emscripten/cache/sysroot/include/compat
0.342  /emsdk/upstream/lib/clang/23/include
0.342  /emsdk/upstream/emscripten/cache/sysroot/include
0.342 End of search list.
0.366  /emsdk/upstream/bin/clang --version
0.376  /emsdk/node/22.16.0_64bit/bin/node /emsdk/upstream/emscripten/tools/compiler.mjs - --symbols-only
0.480  /emsdk/upstream/bin/wasm-ld -o /tmp/hello.wasm /tmp/tmpcdn6uf_dlibemscripten_js_symbols.so --strip-debug --export=add --export=emscripten_stack_get_end --export=emscripten_stack_get_free --export=emscripten_stack_get_base --export=emscripten_stack_get_current --export=emscripten_stack_init --export=_emscripten_stack_alloc --export=__wasm_call_ctors --export=_emscripten_stack_restore --export-if-defined=__start_em_asm --export-if-defined=__stop_em_asm --export-if-defined=__start_em_lib_deps --export-if-defined=__stop_em_lib_deps --export-if-defined=__start_em_js --export-if-defined=__stop_em_js --export-if-defined=main --export-if-defined=__main_argc_argv --export-if-defined=fflush --export-table -z stack-size=65536 --no-growable-memory --initial-heap=16777216 --no-entry --stack-first --table-base=1 /tmp/emscripten_temp_chu_932q/hello.o -L/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten -L/emsdk/upstream/emscripten/src/lib -lGL-getprocaddr -lal -lhtml5 -lstubs-debug -lnoexit -lc-debug -ldlmalloc-debug -lcompiler_rt -lc++-debug-noexcept -lc++abi-debug-noexcept -lsockets -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr
0.504  /emsdk/upstream/bin/llvm-objcopy /tmp/hello.wasm /tmp/hello.wasm '--remove-section=llvm.func_attr.annotate.*' --remove-section=producers --remove-section=name '--remove-section=.debug*'
0.508  /emsdk/node/22.16.0_64bit/bin/node /emsdk/upstream/emscripten/tools/compiler.mjs -
0.606  /emsdk/node/22.16.0_64bit/bin/tsc --outFile /tmp/emscripten_temp_chu_932q/jsdoc.d.ts --skipLibCheck --declaration --emitDeclarationOnly --allowJs /tmp/emscripten_temp_chu_932q/jsdoc.js
0.935 error TS5101: Option 'outFile' is deprecated and will stop functioning in TypeScript 7.0. Specify compilerOption '"ignoreDeprecations": "6.0"' to silence this error.
0.945 emcc: error: '/emsdk/node/22.16.0_64bit/bin/tsc --outFile /tmp/emscripten_temp_chu_932q/jsdoc.d.ts --skipLibCheck --declaration --emitDeclarationOnly --allowJs /tmp/emscripten_temp_chu_932q/jsdoc.js' failed (returned 2)
```

Here is a reproducible Dockerfile:

```docker
FROM emscripten/emsdk:5.0.4

# Install TypeScript 6 globally (the version that breaks --emit-tsd)
RUN npm install -g typescript@6.0.2

# Minimal C source with an exported function
RUN cat > /tmp/hello.c << 'EOF'
#include <emscripten.h>

EMSCRIPTEN_KEEPALIVE
int add(int a, int b) { return a + b; }
EOF

# This should fail: emscripten's --emit-tsd uses `tsc --outFile` internally,
# which was removed in TypeScript 6.
RUN emcc /tmp/hello.c -o /tmp/hello.js \
      -s EXPORTED_FUNCTIONS='["_add"]' \
      -s EXPORTED_RUNTIME_METHODS='["cwrap"]' \
      --emit-tsd /tmp/hello.d.ts \
      -v
```

--- a/tools/emscripten.py
+++ b/tools/emscripten.py
@@ -656,8 +656,8 @@ def create_tsd_exported_runtime_methods(metadata):
           snippet = ' = null'
     js_doc += f'{docs}\nRuntimeExports[\'{name}\']{snippet};\n'
 
-  js_doc_file = in_temp('jsdoc.js')
-  tsc_output_file = in_temp('jsdoc.d.ts')
+  file = 'jsdoc'
+  js_doc_file = in_temp(f'{file}.js')
   utils.write_file(js_doc_file, js_doc)
   tsc = shared.get_npm_cmd('tsc', missing_ok=True)
   # Prefer the npm install'd version of tsc since we know that one is compatible
@@ -669,13 +669,12 @@ def create_tsd_exported_runtime_methods(metadata):
       exit_with_error('tsc executable not found in node_modules or in $PATH')
     # Use the full path from the which command so windows can find tsc.
     tsc = [tsc]
-  cmd = tsc + ['--outFile', tsc_output_file,
-               '--skipLibCheck', # Avoid checking any of the user's types e.g. node_modules/@types.
+  cmd = tsc + ['--skipLibCheck', # Avoid checking any of the user's types e.g. node_modules/@types.
                '--declaration',
                '--emitDeclarationOnly',
                '--allowJs', js_doc_file]
   shared.check_call(cmd, cwd=path_from_root())
-  return utils.read_file(tsc_output_file)
+  return utils.read_file(in_temp(f'{file}.d.ts'))
 
 
 def create_tsd(metadata, embind_tsd):

